### PR TITLE
OCPBUGS-4615: Correct rule description

### PR DESCRIPTION
**Pull request by Vincent056**
Fixes an [issue](https://issues.redhat.com/browse/OCPBUGS-4615) where `ComplianceCheckResult` objects do not have correct descriptions, we corrected the descriptions, and also added a new `rationale` field to `ComplianceCheckResult` objects.

### CODE REVIEW

File: pkg/utils/parse_arf_result.go [hunk 1]
```diff
@@ -552,12 +569,32 @@ func getSafeText(nptr *xmlquery.Node, elem string) string {
...
 }
 
-func complianceCheckResultDescription(rule *xmlquery.Node) string {
+func complianceCheckResultDescription(rule *xmlquery.Node, valuesList map[string]string) (string, error) {
 	title := getSafeText(rule, "xccdf-1.2:title")
 	if title != "" {
 		title = title + "\n"
 	}
-	return title + getSafeText(rule, "xccdf-1.2:rationale")
+	description := rule.SelectElement("xccdf-1.2:description")
```
Comment (jhrozek) [inline pkg/utils/parse_arf_result.go:577]: looks good! Could we reduce code duplication by having a shared function where we would pass the element name and have the shared function be called by both `complianceCheckResultDescription` and `complianceCheckResultRationale`?

File: pkg/utils/parse_arf_result.go [hunk 2]
```diff
@@ -552,12 +569,29 @@ func getSafeText(nptr *xmlquery.Node, elem string) string {
...
+	description, err := getElementText(rule, "xccdf-1.2:description", valuesList)
+	return title + description, err
+}
+
+func complianceCheckResultRationale(rule *xmlquery.Node, valuesList map[string]string) (string, error) {
+	return getElementText(rule, "xccdf-1.2:rationale", valuesList)
+}
+
+func getElementText(nptr *xmlquery.Node, elem string, valuesList map[string]string) (string, error) {
+	ele := nptr.SelectElement("xccdf-1.2:rationale")
```
Comment (jhrozek) [inline pkg/utils/parse_arf_result.go:586]: This should say elem, right?

Comment (Vincent056) [reply pkg/utils/parse_arf_result.go:586]: ahh nice catch, don't know how did I miss that

**Review by jhrozek — APPROVED** [review]
/lgtm
thanks!

**Pull request merged**